src/foam/nanos/crunch/UserCapabilityJunctionDAO.js

```diff
@@ -137,12 +137,27 @@ foam.CLASS({
           if ( capability == null ) {
             logger.error(this.getClass().getSimpleName(), "Capability not found", ucj.getTargetId());
             alarm.setNote(this.ERROR_CAPABILITY_NOT_FOUND + ucj.getTargetId());
-          } else if ( capability.getOf() != null &&
-                      ucj.getData() != null ) {
-            logger.error(this.getClass().getSimpleName(), "Type mismatch", "capability", capability.getId(), "expected", capability.getOf().getId(), "received", ucj.getData().getClassInfo().getId());
           }
-          ((DAO) x.get("alarmDAO")).put(alarm);
-          throw new RuntimeException(alarm.getNote());
+          // want to handle the case where ! ( ucj.getData().getClassInfo().equals(capability.getOf())
+            if ( ! ucj.getData().getClassInfo().equals(capability.getOf()) ) {
+              // I want to then check if class extends ucj.getData()
+              if ( ucj.getData().getClassInfo().getClass().isAssignableFrom(capability.getOf().getClass()) ) {
```

Comment:
Do you need to call getClassInfo()? Can't you just call getClass() on the object directly?

Reply:
like this: ? ucj.getData.getClass() == capability.getOf().getClass()